### PR TITLE
Added a getStatusCode method.

### LINKED ISSUE #27: Could we have a method to go from status text to status code?
For instance, I need to be able to pass in the text 'IM_A_TEAPOT' and get out 418.

Currently 'getStatusText' does the opposite (pass in 418 and get IM_A_TEAPOT out), and I can do this (hacky) in the meantime:

```
import codes from 'http-status-codes'

console.log(codes['IM_A_TEAPOT'])
```

But this doesn't seem ideal.

### PR DESCRIPTION
**Pull request by o0Ignition0o**
this should fix #27 

Please let me know if there's anything I can do to improve the pull request :)

### CODE REVIEW

**Comment by luzlab** [comment]
Can we please merge this or PR29 to get the typescript enums?

**Comment by prettymuchbryce** [comment]
My only issue with this change is that currently this project is not compiled. This means using features like `Array.find` and `Object.entries` without a polyfill could break browser compatibility.

https://caniuse.com/#feat=array-find

**Comment by o0Ignition0o** [comment]
Oh I understand, do you think using a for loop instead is the way to go ? if so i can edit it today :)

**Comment by o0Ignition0o** [comment]
Using a for loop + rebased as required :)

**Comment by prettymuchbryce** [comment]
This looks better. We still need to add this to the TS definition file, and probably some mention in the README as well.

**Comment by o0Ignition0o** [comment]
Oh yeah sure, I will add it today. Thanks a lot for providing a bit of
mentoring despite a tight schedule !

File: index.js [hunk 1]
```diff
@@ -74,3 +74,13 @@ exports.getStatusText = function(statusCode) {
     throw new Error("Status code does not exist: " + statusCode);
   }
 };
+
+exports.getStatusCode = function(statusText) {
+  for (key in statusCodes) {
+    if (statusCodes[key] === statusText) {
+      return parseInt(key, 10);
```
Comment (o0Ignition0o) [inline index.js]: Do we want to be more defensive here and check if parseInt is a number?

Comment (prettymuchbryce) [reply index.js]: I think it's fine. Lets assume `key` is always some Number (status code) coerced into a string.

File: README.md [hunk 2]
```diff
@@ -26,6 +26,9 @@ response
 	.send({
 		error: HttpStatus.getStatusText(HttpStatus.INTERNAL_SERVER_ERROR)
 	});
+
+response.status(HttpStatus.getStatusCode('No Content'))
```
Comment (prettymuchbryce) [inline README.md]: Can you format these with the line breaks like the others?

File: README.md [hunk 3]
```diff
@@ -107,14 +110,22 @@ response
 	.send({
 		error: HttpStatus.getStatusText(HttpStatus.INTERNAL_SERVER_ERROR)
 	})
+
+response
+	.status(getStatusCode('No Content')
```
Comment (prettymuchbryce) [inline README.md]: How about we just use internal server error like the example above?

File: index.d.ts [hunk 4]
```diff
@@ -384,5 +384,13 @@ export declare enum HTTP_STATUS {
  * Convert the numeric status code to its appropriate title.
  * @param statusCode One of the available status codes in this package
  * @returns {String} The associated title of the passed status code
+ * @throws {Error} The status code does not exist
```
Comment (prettymuchbryce) [inline index.d.ts]: Nice catch. 👍

File: index.d.ts [hunk 5]
```diff
@@ -384,5 +384,13 @@ export declare enum HTTP_STATUS {
...
  */
 export declare function getStatusText(statusCode: number): string;
+
+/**
+ * Convert the status code title to its appropriate numeric value
+ * @param statusText One of the available status texts in this package
+ * @returns {Number} The associated status code of the passed status text
+ * @throws {Error} The status text does not exist
```
Comment (prettymuchbryce) [inline index.d.ts]: I think you forgot to close this comment.

File: index.d.ts [hunk 6]
```diff
@@ -384,5 +384,13 @@ export declare enum HTTP_STATUS {
  * Convert the numeric status code to its appropriate title.
  * @param statusCode One of the available status codes in this package
  * @returns {String} The associated title of the passed status code
+ * @throws {Error} The status code does not exist
  */
 export declare function getStatusText(statusCode: number): string;
+
+/**
+ * Convert the status code title to its appropriate numeric value
```
Comment (prettymuchbryce) [inline index.d.ts]: I have some problem with the word "title" in these comments. We should refer to them consistently as "status code" and "status text". 

In fact the IETF spec calls it a "Reason Phrase" so ideally we could change all of the comments from "title" -> "reason phrase".

```
The reason-phrase element exists for the sole purpose of providing a
   textual description associated with the numeric status code, mostly
   out of deference to earlier Internet application protocols that were
   more frequently used with interactive text clients.  A client SHOULD
   ignore the reason-phrase content.
```

**Comment by o0Ignition0o** [comment]
Thanks a lot for such a thorough code review. I think I have addressed most of the concerns :)

By the way I was skimming over the HTTP status codes and it seems like the 500 Reason phrase is ["Internal Server Error"](https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.5.1) whereas we here refer to it as ["Server Error"](https://github.com/prettymuchbryce/http-status-codes/blob/master/README.md#codes).

As stated in the IETF spec, a client should ignore the reason-phrasse content so it's probably not a big deal, plus we probably wouldn't want to cause a breaking change ^^

**Comment by justinmchase** [comment]
Can we merge this now please?

**Comment by prettymuchbryce** [comment]
Sorry for the delay.

**Pull request merged**